### PR TITLE
update changelog

### PR DESCRIPTION
**Pull request by rfomin**
I think I missed a lot of HUD changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,39 +1,74 @@
...
-* Attempt to play `demo4` also for Doom 2 (Final Doom).
-* Clean screenshots are taken with the base palette.
+* Implement support for arbitrary resolutions.
+  - Resolution scaling slider. From 100% (original resolution 320x200) and up to native display resolution.
+  - Dynamic resolution scaling option. Automatically changes the internal resolution to maintain target frame rate (display refresh rate by default).
+  - "Blocky Fuzz" adapted to any resolution. Fixed some visual artefacts.
+  - Vanilla wipe at any resolution (from Diet Boom).
+* Add adjustable FOV. This uses a 4:3 horizontal FOV and automatically adjusts it to the current aspect ratio. That means most users should leave the FOV at 90 (default) unless they have a specific reason to change it.
+* Integration of Andrew Apted voxel code. Support for models in KVX format. Recommended mod: [Voxel Doom II v2.3](https://www.moddb.com/mods/voxel-doom-ii/addons/voxel-doom-ii-with-parallax-textures). Just run ZIP/PK3 file with `-file` parameter or place it in autoload folder.
+* Mouse controls overhaul.
```
[the remark this answers is not in the document]
Comment (rfomin) [reply CHANGELOG.md:9]: Done in https://github.com/fabiangreffrath/woof/pull/1442/commits/f5b772038330fd7e5cd4cd173da4c339e796166b

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,39 +1,74 @@
...
+  - Fast mouse polling. Instead of sampling the mouse input once per tic, it's now sampled once per frame. This lowers the input lag considerably at high frame rates, especially with a high refresh rate monitor.
```
[the remark this answers is not in the document]
Comment (rfomin) [reply CHANGELOG.md]: Sure. Feel free to edit this file as you like (I think it's even possible to do this directly from the GitHub UI). I will appreciate wording/grammar corrections.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,39 +1,74 @@
...
-* Allow separate key binding for the numeric keypad.
-* Replace and extend crosshair patches with the shaded variants from Nugget Doom.
-* Ignore DMX sound padding (@ceski-1).
-* Implement sky top color algorithm from Eternity Engine (@tomas7770).
-* Attempt to play `demo4` also for Doom 2 (Final Doom).
-* Clean screenshots are taken with the base palette.
+* Implement support for arbitrary resolutions.
+  - Resolution scaling slider. From 100% (original resolution 320x200) and up to native display resolution.
+  - Dynamic resolution scaling option. Automatically changes the internal resolution to maintain target frame rate (display refresh rate by default).
+  - "Blocky Fuzz" adapted to any resolution. Fixed some visual artefacts.
```
[the remark this answers is not in the document]
Comment (rfomin) [reply CHANGELOG.md:5]: Could you please point out the problems? I tried fuzztest.wad and it looks OK to me.

[the remark this answers is not in the document]
Comment (ceski-1) [reply CHANGELOG.md:5]: > Could you please point out the problems? I tried fuzztest.wad and it looks OK to me.

Fuzz hasn't look right at some point after this commit: https://github.com/fabiangreffrath/woof/pull/1217

Try a few different resolutions while looking at the test pattern that replaces the pistol. Then take a look at the fist and how it looks when it's returning to an idle state (it shrinks in a strange manner). Even without the fuzz effect, player sprites look a little odd when returning to an idle state.

[the remark this answers is not in the document]
Comment (rfomin) [reply CHANGELOG.md:5]: > Then take a look at the fist and how it looks when it's returning to an idle state after coming to a stop (it shrinks in a strange manner). Even without the fuzz effect, player sprites look a little odd when returning to an idle state (move around a bit and then let the weapon sprite return to center).

I think it's the aftermath of <https://github.com/fabiangreffrath/woof/pull/1377> Perhaps we should revert it.

**Comment by fabiangreffrath** [comment]
> I think I missed a lot of HUD changes.

The only thing that's missing is that the IDRATE cheat now has its own widget.

**Comment by rfomin** [comment]
> The only thing that's missing is that the IDRATE cheat now has its own widget.

I think I should update it, add colors and put it on the second line. It seems that this widget will be popular.

**Pull request merged**